app/src/main/java/org/schabi/newpipe/player/helper/PlayerHolder.java

```diff
@@ -85,58 +104,65 @@ public static void startService(final Context context,
         // and NullPointerExceptions inside the service because the service will be
         // bound twice. Prevent it with unbinding first
         unbind(context);
-        context.startService(new Intent(context, MainPlayer.class));
-        serviceConnection = getServiceConnection(context, playAfterConnect);
+        ContextCompat.startForegroundService(context, new Intent(context, MainPlayer.class));
+        serviceConnection.doPlayAfterConnect(playAfterConnect);
         bind(context);
     }
 
-    public static void stopService(final Context context) {
+    public void stopService() {
+        final Context context = getCommonContext();
         unbind(context);
         context.stopService(new Intent(context, MainPlayer.class));
     }
 
-    private static ServiceConnection getServiceConnection(final Context context,
-                                                          final boolean playAfterConnect) {
-        return new ServiceConnection() {
-            @Override
-            public void onServiceDisconnected(final ComponentName compName) {
-                if (DEBUG) {
-                    Log.d(TAG, "Player service is disconnected");
-                }
+    class PlayerServiceConnection implements ServiceConnection {
+
+        private boolean playAfterConnect = false;
+
+        public void doPlayAfterConnect(final boolean playAfterConnection) {
+            this.playAfterConnect = playAfterConnection;
```

Comment:
```suggestion
        public void doPlayAfterConnect(final boolean playAfterConnect) {
            this.playAfterConnect = playAfterConnect;
```

Reply:
This one was strange! I had it just like you suggested but checkStyle did not want to let me go. I'll recheck again

Reply:
Oh, then leave it as it is, nevermind